File: CS/GridControlCRUDMVVMInfiniteAsyncSource/IssueDataFilterConverter.cs
using DevExpress.Data.Filtering;
using DevExpress.Xpf.Data;

namespace GridControlCRUDMVVMInfiniteAsyncSource {
    public class IssueDataFilterConverter : ExpressionFilterConverter<IssueData> {
        protected override void SetUpConverter(GridFilterCriteriaToExpressionConverter<IssueData> converter) {
            converter.RegisterFunctionExpressionFactory(
                operatorType: FunctionOperatorType.StartsWith,
                factory: (string value) => {
                    var toLowerValue = value.ToLower();
                    return x => x.ToLower().StartsWith(toLowerValue);
                });
        }
    }
}

File: VB/GridControlCRUDMVVMInfiniteAsyncSource/IssueDataFilterConverter.vb
Imports DevExpress.Data.Filtering
Imports DevExpress.Xpf.Data

Namespace GridControlCRUDMVVMInfiniteAsyncSource
	Public Class IssueDataFilterConverter
		Inherits ExpressionFilterConverter(Of IssueData)

		Protected Overrides Sub SetUpConverter(ByVal converter As GridFilterCriteriaToExpressionConverter(Of IssueData))
			converter.RegisterFunctionExpressionFactory(operatorType:= FunctionOperatorType.StartsWith, factory:= Function(value As String)
				Dim toLowerValue = value.ToLower()
				Return x
				If True Then
					Get
						Return x.ToLower().StartsWith(toLowerValue)
					End Get
				End If
			End Function)
		End Sub
	End Class
End Namespace
